### ISSUE TITLE
Anonymous class variable definition inside method call arguments not propagating

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#7762]
https://psalm.dev/r/941c731b3a

**Comment by orklah** [comment]
Yeah, right now, an anonymous class in Psalm can only have an supertype when it's extended. Attributes and methods are not checked.

I'm not personally interested in working on this for the moment, if someone is, I could give a few pointers.

A first (probably easy) step would be to consider any anonymous class as valid in a callable array. The part about the property and checking whether `test` is actually a valid method would be harder.

A workaround for this could be to create an actual class with attributes and methods and then creating an anonymous class that just extends the actual class. Not sure about how it would defeat the purpose of this code though.
